ui/src/containers/configs/EditConfig.js

```diff
@@ -356,612 +356,703 @@ class EditConfig extends Component {
+                            value.target.value,
+                            "deployment.spec.replicas"
+                          );
+                        }}
+                        placeholder="1"
+                        step="1"
+                        value={
+                          !isNaN(this.state.config.spec["replicas"])
+                            ? this.state.config.spec["replicas"]
+                            : ""
+                        }
+                      />
+                    </div>
+                  </>
                 )}
-                <div className="col-12 mt-4">
-                  <a
-                    href={`/configs/${config.uuid}`}
-                    className="btn btn-primary text-white mb-4"
-                    onClick={this.handleUpdateConfig}
-                  >
-                    Update config
-                  </a>
-                </div>
+              </>
+            )}
+            {![undefined, ""].includes(this.state.errorMessage) && (
+              <div className="alert alert-danger mt-4">
+                <p className="h6 fs-bolder">API response:</p>
+                {this.state.errorMessage}
               </div>
+            )}
+            <div className="col-12 mt-4 mb-4">
+              <a
+                href={`/configs/${config.uuid}`}
+                className="btn btn-primary text-white"
+                onClick={this.handleUpdateConfig}
+              >
+                Update config
+              </a>
+              <button
+                className="btn btn-outline-primary ms-2"
+                onClick={this.toggleForm}
+              >
+                Edit as JSON
+              </button>
             </div>
-          </form>
+          </div>
+        </div>
+      </form>
+    );
+  }
+
+  toggleForm(event) {
+    event.preventDefault();
+    let isShowingPayloadEditor = !this.state.showPayloadEditor;
+
+    if (isShowingPayloadEditor) {
+      this.setState({
+        showPayloadEditor: isShowingPayloadEditor,
+        editorConfig: JSON.stringify(this.state.config, null, 2),
+      });
+    } else if (
+      this.state.payloadEditorChanges &&
+      !window.confirm("Going back will reset all edits in the editor!")
+    ) {
+      this.setState({
+        showPayloadEditor: true,
+      });
+    } else {
+      this.setState({
+        editorConfig: JSON.stringify(this.state.config, null, 2),
+        showPayloadEditor: isShowingPayloadEditor,
+        payloadEditorChanges: false,
+        errorMessage: "",
+      });
+    }
+  }
+
+  submitEditorContent(event) {
+    event.preventDefault();
+    let parsedConfig = undefined;
+    try {
+      parsedConfig = JSON.parse(this.state.editorConfig, null, 2);
+    } catch (syntaxError) {
+      this.setState({
+        configCreated: false,
+        errorMessage: syntaxError.message,
+      });
+    }
+
+    if (parsedConfig !== undefined) {
+      this.props.updateConfig(parsedConfig).then(() => {
```

Comment:
This function [expects two parameter](https://github.com/DataCater/datacater/blob/main/ui/src/actions/configs.js#L108). This way it is not working.